pkg/tagger/collectors/workloadmeta_extract.go

```diff
@@ -222,6 +222,14 @@ func (c *WorkloadMetaCollector) handleContainer(ev workloadmeta.Event) []*TagInf
 		utils.AddMetadataAsTags(envName, envValue, c.containerEnvAsTags, c.globContainerEnvLabels, tags)
 	}
 
+	// tags from the pod
+	pod, err := c.store.GetKubernetesPodForContainer(container.ID)
```

Comment:
Kubernetes pods get kubernetes tags from `handleKubePod` already, there's no need to fetch the pod here, the rest of the code takes care of it already.